packages/keyring-controller/src/KeyringController.ts

```diff
@@ -312,7 +315,7 @@ export class KeyringController extends BaseController<
       } else {
         vault = await this.#keyring.createNewVaultAndKeychain(password);
         this.updateIdentities(await this.getAccounts());
-        this.fullUpdate();
+        await this.fullUpdate();
```

Comment:
This missing await made the KeyringController state out of sync when calling `createNewVaultAndKeychain`